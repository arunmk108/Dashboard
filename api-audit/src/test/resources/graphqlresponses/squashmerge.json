{
    "data": {
        "repository": {
            "pullRequests": {
                "totalCount": 1,
                "pageInfo": {
                    "endCursor": "Y3Vyc29yOnYyOpK5MjAxNy0xMi0xMlQxMzoxMjo0My0wNTowMM4ADTnk",
                    "hasNextPage": false
                },
                "edges": [
                    {
                        "cursor": "Y3Vyc29yOnYyOpK5MjAxNy0xMi0xMlQxMzoxMjo0My0wNTowMM4ADTnk",
                        "node": {
                            "number": 1,
                            "mergeable": "UNKNOWN",
                            "state": "MERGED",
                            "createdAt": "2017-12-12T18:12:29Z",
                            "updatedAt": "2017-12-12T18:12:43Z",
                            "mergedAt": "2017-12-12T18:12:43Z",
                            "baseRef": {
                                "target": {
                                    "oid": "0a9404e92b613c30e7a30606faf4cc1e69fd684d"
                                }
                            },
                            "baseRefName": "master",
                            "title": "pr from fork",
                            "mergeCommit": {
                                "oid": "451145cb358eb78b9d2e12051c48fe8d5219ba1e"
                            },
                            "headRepository": {
                                "name": "SquashMerge",
                                "url": "https://mygithub.com/ghtwo/SquashMerge"
                            },
                            "headRef": {
                                "target": {
                                    "oid": "2df478075f504813ec52a0750a5ab58aa06a543d"
                                }
                            },
                            "headRefName": "master",
                            "author": {
                                "login": "ghtwo",
                                "resourcePath": "/ghtwo"
                            },
                            "commits": {
                                "totalCount": 2,
                                "nodes": [
                                    {
                                        "commit": {
                                            "oid": "abdca9410a861d28ee0772433c1946d066577454",
                                            "parents": {
                                                "nodes": [
                                                    {
                                                        "oid": "b45d5a45765b83886e953ac4198d10145a42bdf4"
                                                    }
                                                ]
                                            },
                                            "committedDate": "2017-12-12T18:12:00Z",
                                            "message": "Third fork commit",
                                            "status": null,
                                            "author": {
                                                "name": "Author, Two",
                                                "date": "2017-12-12T13:12:00-05:00",
                                                "user": {
                                                    "login": "ghtwo"
                                                }
                                            }
                                        }
                                    },
                                    {
                                        "commit": {
                                            "oid": "2df478075f504813ec52a0750a5ab58aa06a543d",
                                            "parents": {
                                                "nodes": [
                                                    {
                                                        "oid": "abdca9410a861d28ee0772433c1946d066577454"
                                                    }
                                                ]
                                            },
                                            "committedDate": "2017-12-12T18:12:13Z",
                                            "message": "Fourth fork commit",
                                            "status": null,
                                            "author": {
                                                "name": "Author, Two",
                                                "date": "2017-12-12T13:12:13-05:00",
                                                "user": {
                                                    "login": "ghtwo"
                                                }
                                            }
                                        }
                                    }
                                ]
                            },
                            "comments": {
                                "totalCount": 0,
                                "nodes": []
                            },
                            "reviews": {
                                "totalCount": 0,
                                "nodes": []
                            }
                        }
                    }
                ]
            },
            "ref": {
                "target": {
                    "history": {
                        "pageInfo": {
                            "endCursor": "0a9404e92b613c30e7a30606faf4cc1e69fd684d 4",
                            "hasNextPage": false
                        },
                        "edges": [
                            {
                                "cursor": "0a9404e92b613c30e7a30606faf4cc1e69fd684d 0",
                                "node": {
                                    "oid": "0a9404e92b613c30e7a30606faf4cc1e69fd684d",
                                    "parents": {
                                        "nodes": [
                                            {
                                                "oid": "01be3afcdf98ce45a88897dfc5a3d485b9c17ce6"
                                            }
                                        ]
                                    },
                                    "message": "Fifth direct commit",
                                    "committer": {
                                        "user": null,
                                        "name": "GitHub Enterprise",
                                        "date": "2017-12-12T13:13:38-05:00"
                                    },
                                    "author": {
                                        "name": "Author, Two",
                                        "user": {
                                            "login": "ghtwo"
                                        },
                                        "email": "Author.Two@mycompany.com",
                                        "date": "2017-12-12T13:13:38-05:00"
                                    },
                                    "status": null
                                }
                            },
                            {
                                "cursor": "0a9404e92b613c30e7a30606faf4cc1e69fd684d 1",
                                "node": {
                                    "oid": "01be3afcdf98ce45a88897dfc5a3d485b9c17ce6",
                                    "parents": {
                                        "nodes": [
                                            {
                                                "oid": "451145cb358eb78b9d2e12051c48fe8d5219ba1e"
                                            }
                                        ]
                                    },
                                    "message": "Fourth direct commit",
                                    "committer": {
                                        "user": null,
                                        "name": "GitHub Enterprise",
                                        "date": "2017-12-12T13:13:25-05:00"
                                    },
                                    "author": {
                                        "name": "Author, Two",
                                        "user": {
                                            "login": "ghtwo"
                                        },
                                        "email": "Author.Two@mycompany.com",
                                        "date": "2017-12-12T13:13:25-05:00"
                                    },
                                    "status": null
                                }
                            },
                            {
                                "cursor": "0a9404e92b613c30e7a30606faf4cc1e69fd684d 2",
                                "node": {
                                    "oid": "451145cb358eb78b9d2e12051c48fe8d5219ba1e",
                                    "parents": {
                                        "nodes": [
                                            {
                                                "oid": "b45d5a45765b83886e953ac4198d10145a42bdf4"
                                            }
                                        ]
                                    },
                                    "message": "pr from fork (#1)\n\n* Third fork commit\r\n\r\n* Fourth fork commit",
                                    "committer": {
                                        "user": null,
                                        "name": "GitHub Enterprise",
                                        "date": "2017-12-12T13:12:42-05:00"
                                    },
                                    "author": {
                                        "name": "Author, Two",
                                        "user": {
                                            "login": "ghtwo"
                                        },
                                        "email": "Author.Two@mycompany.com",
                                        "date": "2017-12-12T13:12:42-05:00"
                                    },
                                    "status": null
                                }
                            },
                            {
                                "cursor": "0a9404e92b613c30e7a30606faf4cc1e69fd684d 3",
                                "node": {
                                    "oid": "b45d5a45765b83886e953ac4198d10145a42bdf4",
                                    "parents": {
                                        "nodes": [
                                            {
                                                "oid": "74398c71c338a326af27efa0bb66153f0e8c6617"
                                            }
                                        ]
                                    },
                                    "message": "Second direct commit",
                                    "committer": {
                                        "user": null,
                                        "name": "GitHub Enterprise",
                                        "date": "2017-12-12T13:11:31-05:00"
                                    },
                                    "author": {
                                        "name": "Author, Two",
                                        "user": {
                                            "login": "ghtwo"
                                        },
                                        "email": "Author.Two@mycompany.com",
                                        "date": "2017-12-12T13:11:31-05:00"
                                    },
                                    "status": null
                                }
                            },
                            {
                                "cursor": "0a9404e92b613c30e7a30606faf4cc1e69fd684d 4",
                                "node": {
                                    "oid": "74398c71c338a326af27efa0bb66153f0e8c6617",
                                    "parents": {
                                        "nodes": []
                                    },
                                    "message": "First direct commit",
                                    "committer": {
                                        "user": null,
                                        "name": "GitHub Enterprise",
                                        "date": "2017-12-12T13:11:14-05:00"
                                    },
                                    "author": {
                                        "name": "Author, Two",
                                        "user": {
                                            "login": "ghtwo"
                                        },
                                        "email": "Author.Two@mycompany.com",
                                        "date": "2017-12-12T13:11:14-05:00"
                                    },
                                    "status": null
                                }
                            }
                        ]
                    }
                }
            },
            "issues": {
                "totalCount": 0,
                "pageInfo": {
                    "endCursor": null,
                    "hasNextPage": false
                },
                "edges": []
            }
        }
    }
}