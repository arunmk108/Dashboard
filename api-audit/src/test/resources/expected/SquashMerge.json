[
    {
        "auditStatuses": [
            "GIT_FORK_STRATEGY",
            "COMMITAUTHOR_EQ_MERGECOMMITER",
            "PULLREQ_NOT_PEER_REVIEWED"
        ],
        "lastUpdated": 1516207280261,
        "pullRequest": {
            "scmUrl": "https://mygithub.com/Devopscode/SquashMerge",
            "scmBranch": "master",
            "scmRevisionNumber": "451145cb358eb78b9d2e12051c48fe8d5219ba1e",
            "scmCommitLog": "pr from fork",
            "scmCommitTimestamp": 1513102363000,
            "numberOfChanges": 0,
            "id": "5a5f77a16fa3ba5285b9753f",
            "orgName": "Devopscode",
            "repoName": "SquashMerge",
            "sourceRepo": "ghtwo/SquashMerge",
            "sourceBranch": "master",
            "targetRepo": "Devopscode/SquashMerge",
            "targetBranch": "master",
            "number": "1",
            "collectorItemId": "5a4e54ba33ec296b81c39390",
            "updatedAt": 1513102363000,
            "createdAt": 1513102349000,
            "closedAt": 0,
            "state": "merged",
            "mergedAt": 1513102363000,
            "timestamp": 1513102349000,
            "resolutiontime": 0,
            "userId": "ghtwo",
            "comments": [],
            "reviews": [],
            "headSha": "2df478075f504813ec52a0750a5ab58aa06a543d",
            "baseSha": "0a9404e92b613c30e7a30606faf4cc1e69fd684d",
            "requestType": "pull",
            "commits": [
                {
                    "scmRevisionNumber": "abdca9410a861d28ee0772433c1946d066577454",
                    "scmCommitLog": "Third fork commit",
                    "scmAuthor": "Author, Two",
                    "scmAuthorLogin": "ghtwo",
                    "scmParentRevisionNumbers": [
                        "b45d5a45765b83886e953ac4198d10145a42bdf4"
                    ],
                    "scmCommitTimestamp": 1513102320000,
                    "numberOfChanges": 0,
                    "timestamp": 0,
                    "firstEverCommit": false
                },
                {
                    "scmRevisionNumber": "2df478075f504813ec52a0750a5ab58aa06a543d",
                    "scmCommitLog": "Fourth fork commit",
                    "scmAuthor": "Author, Two",
                    "scmAuthorLogin": "ghtwo",
                    "scmParentRevisionNumbers": [
                        "abdca9410a861d28ee0772433c1946d066577454"
                    ],
                    "scmCommitTimestamp": 1513102333000,
                    "numberOfChanges": 0,
                    "timestamp": 0,
                    "firstEverCommit": false
                }
            ]
        },
        "commits": [
            {
                "scmRevisionNumber": "abdca9410a861d28ee0772433c1946d066577454",
                "scmCommitLog": "Third fork commit",
                "scmAuthor": "Author, Two",
                "scmAuthorLogin": "ghtwo",
                "scmParentRevisionNumbers": [
                    "b45d5a45765b83886e953ac4198d10145a42bdf4"
                ],
                "scmCommitTimestamp": 1513102320000,
                "numberOfChanges": 0,
                "timestamp": 0,
                "firstEverCommit": false
            },
            {
                "scmRevisionNumber": "2df478075f504813ec52a0750a5ab58aa06a543d",
                "scmCommitLog": "Fourth fork commit",
                "scmAuthor": "Author, Two",
                "scmAuthorLogin": "ghtwo",
                "scmParentRevisionNumbers": [
                    "abdca9410a861d28ee0772433c1946d066577454"
                ],
                "scmCommitTimestamp": 1513102333000,
                "numberOfChanges": 0,
                "timestamp": 0,
                "firstEverCommit": false
            }
        ],
        "scmUrl": "https://mygithub.com/Devopscode/SquashMerge",
        "scmBranch": "master"
    },
    {
        "auditStatuses": [
            "SCM_AUTHOR_LOGIN_INVALID",
            "DIRECT_COMMITS_TO_BASE_FIRST_COMMIT",
            "DIRECT_COMMITS_TO_BASE"
        ],
        "lastUpdated": 1516207280261,
        "commits": [
            {
                "scmUrl": "https://mygithub.com/Devopscode/SquashMerge",
                "scmBranch": "master",
                "scmRevisionNumber": "0a9404e92b613c30e7a30606faf4cc1e69fd684d",
                "scmCommitLog": "Fifth direct commit",
                "scmAuthor": "Author, Two",
                "scmAuthorLogin": "ghtwo",
                "scmParentRevisionNumbers": [
                    "01be3afcdf98ce45a88897dfc5a3d485b9c17ce6"
                ],
                "scmCommitTimestamp": 1513102418000,
                "numberOfChanges": 1,
                "type": "New",
                "id": "5a5f77a16fa3ba5285b9753a",
                "collectorItemId": "5a4e54ba33ec296b81c39390",
                "timestamp": 1516205985259,
                "firstEverCommit": false
            },
            {
                "scmUrl": "https://mygithub.com/Devopscode/SquashMerge",
                "scmBranch": "master",
                "scmRevisionNumber": "01be3afcdf98ce45a88897dfc5a3d485b9c17ce6",
                "scmCommitLog": "Fourth direct commit",
                "scmAuthor": "Author, Two",
                "scmAuthorLogin": "ghtwo",
                "scmParentRevisionNumbers": [
                    "451145cb358eb78b9d2e12051c48fe8d5219ba1e"
                ],
                "scmCommitTimestamp": 1513102405000,
                "numberOfChanges": 1,
                "type": "New",
                "id": "5a5f77a16fa3ba5285b9753b",
                "collectorItemId": "5a4e54ba33ec296b81c39390",
                "timestamp": 1516205985259,
                "firstEverCommit": false
            },
            {
                "scmUrl": "https://mygithub.com/Devopscode/SquashMerge",
                "scmBranch": "master",
                "scmRevisionNumber": "b45d5a45765b83886e953ac4198d10145a42bdf4",
                "scmCommitLog": "Second direct commit",
                "scmAuthor": "Author, Two",
                "scmAuthorLogin": "ghtwo",
                "scmParentRevisionNumbers": [
                    "74398c71c338a326af27efa0bb66153f0e8c6617"
                ],
                "scmCommitTimestamp": 1513102291000,
                "numberOfChanges": 1,
                "type": "New",
                "id": "5a5f77a16fa3ba5285b9753d",
                "collectorItemId": "5a4e54ba33ec296b81c39390",
                "timestamp": 1516205985259,
                "firstEverCommit": false
            },
            {
                "scmUrl": "https://mygithub.com/Devopscode/SquashMerge",
                "scmBranch": "master",
                "scmRevisionNumber": "74398c71c338a326af27efa0bb66153f0e8c6617",
                "scmCommitLog": "First direct commit",
                "scmAuthor": "Author, Two",
                "scmAuthorLogin": "ghtwo",
                "scmParentRevisionNumbers": [],
                "scmCommitTimestamp": 1513102274000,
                "numberOfChanges": 1,
                "type": "New",
                "id": "5a5f77a16fa3ba5285b9753e",
                "collectorItemId": "5a4e54ba33ec296b81c39390",
                "timestamp": 1516205985259,
                "firstEverCommit": true
            }
        ],
        "scmUrl": "https://mygithub.com/Devopscode/SquashMerge",
        "scmBranch": "master"
    }
]